flink-yarn-tests/src/test/java/org/apache/flink/yarn/YarnTestBase.java

```diff
@@ -592,14 +608,14 @@ protected ApplicationReport getOnlyApplicationReport() throws IOException, YarnE
 	}
 
 	public static void startYARNSecureMode(YarnConfiguration conf, String principal, String keytab) {
-		start(conf, principal, keytab);
+		start(conf, principal, keytab, false);
 	}
 
-	public static void startYARNWithConfig(YarnConfiguration conf) {
-		start(conf, null, null);
+	public static void startYARNWithConfig(YarnConfiguration conf, boolean withDFS) {
+		start(conf, null, null, withDFS);
 	}
 
-	private static void start(YarnConfiguration conf, String principal, String keytab) {
+	private static void start(YarnConfiguration conf, String principal, String keytab, boolean withDFS) {
```

Comment:
Could you instead use an external configuration code than such boolean flag? Or at least use an overload method so that we don't modify all of these method calls which only pass `withDFS == false`.

Reply:
+1 to Tison's idea. whatsmore, I don't think we need to change the API here. since we can just check whether `FILE_REPLICATION` has been set in Flink Configuration. or whether `DFSConfigKeys.DFS_REPLICATION_KEY` has been set in YarnConfiguration